pandas/_libs/groupby.pyx

```diff
@@ -1375,28 +1379,44 @@ cdef cummin_max(groupby_t[:, ::1] out,
     """
     cdef:
         Py_ssize_t i, j, N, K
-        groupby_t val, mval
+        groupby_t val, mval, na_val
+        uint8_t[::1] seen_na
         intp_t lab
 
+    if groupby_t is float64_t or groupby_t is float32_t:
+        na_val = NaN
+    elif is_datetimelike:
+        na_val = NPY_NAT
+    # Will never be used, just to avoid uninitialized warning
+    else:
+        na_val = 0
+
+    N, K = (<object>values).shape
+    seen_na = np.zeros((<object>accum).shape[0], dtype=np.uint8)
```

Comment:
can we avoid allocating in some cases?

Reply:
Have changed to only allocate if a missing value is possible. Could avoid more aggressively perhaps (eg only as soon as a missing value is seen), see comment for the masked case